File: src/samples/java/samples/LuaThread.java
package samples;


import shibaInu.tolua.LuaState;

import java.util.Random;


/**
 * lua state 在多线程环境下运行
 */
public class LuaThread extends Thread {


    public LuaThread(String name) {
        this.setName(name);
    }

    @Override
    public void run() {

        LuaState state = new LuaState();

        state.doFile("LuaThread");
        state.doString(String.format("SetName('%s')", getName()));

        for (int i = 0; i < 30; i++) {
            try {
                Thread.sleep(Math.abs(new Random().nextLong() % 200));
                state.doString("Next()");
            } catch (Exception e) {
                System.out.println(e);
            }
        }

        System.out.println(getName() + "    done.");
        state.destroy();
    }


    //
}

File: src/samples/lua/LuaThread.lua
--



-- test require
local n = 0
n = n + require("Abc.Xyz.N1")
n = n + require("Abc.Xyz.N2")
n = n + require("Abc.Xyz.N3")

local name
local step = 0


--
function SetName(value)
    name = value
    print("LuaThread.lua", name, "n ".. n)
end


--
function Next()
    step = step + 1
    print(name, "step " .. step)
end


